trino/sqlalchemy/dialect.py

```diff
@@ -0,0 +1,297 @@
+        """Trino has no support for check constraints. Returns an empty list."""
+        return []
+
+    def get_table_comment(self, connection: Connection,
+                          table_name: str, schema: str = None, **kw) -> Dict[str, Any]:
+        properties_table = self._get_full_table(f'{table_name}$properties', schema)
+        query = f'SELECT "comment" FROM {properties_table}'
+        try:
+            res = connection.execute(sql.text(query))
+            return dict(text=res.scalar())
+        except error.TrinoQueryError as e:
+            if e.error_name in (
+                    error.NOT_FOUND,
+                    error.COLUMN_NOT_FOUND,
+                    error.TABLE_NOT_FOUND,
+            ):
+                return dict(text=None)
+            raise
+
+    def has_schema(self, connection: Connection, schema: str) -> bool:
+        query = dedent('''
+            SELECT "schema_name"
+            FROM "information_schema"."schemata"
+            WHERE "schema_name" = :schema
+        ''').strip()
+        res = connection.execute(sql.text(query), schema=schema)
+        return res.first() is not None
+
+    def has_table(self, connection: Connection,
+                  table_name: str, schema: str = None, **kw) -> bool:
+        schema = schema or self._get_default_schema_name(connection)
+        if schema is None:
+            return False
+        query = dedent('''
+            SELECT "table_name"
+            FROM "information_schema"."tables"
+            WHERE "table_schema" = :schema
+              AND "table_name" = :table
+        ''').strip()
+        res = connection.execute(sql.text(query), schema=schema, table=table_name)
+        return res.first() is not None
+
+    def has_sequence(self, connection: Connection,
+                     sequence_name: str, schema: str = None, **kw) -> bool:
+        """Trino has no support for sequence. Returns False indicate that given sequence does not exists."""
+        return False
+
+    def _get_server_version_info(self, connection: Connection) -> Tuple[int, ...]:
+        query = 'SELECT version()'
+        res = connection.execute(sql.text(query))
+        version = res.scalar()
+        return tuple([version])
+
+    def _get_default_schema_name(self, connection: Connection) -> Optional[str]:
+        dbapi_connection: trino_dbapi.Connection = connection.connection
+        return dbapi_connection.schema
+
+    def do_execute(self, cursor: Cursor, statement: str, parameters: Tuple[Any, ...],
+                   context: DefaultExecutionContext = None):
+        cursor.execute(statement, parameters)
+        if context and context.should_autocommit:
+            # SQL statement only submitted to Trino server when cursor.fetch*() is called.
+            # For DDL (CREATE/ALTER/DROP) and DML (INSERT/UPDATE/DELETE) statement, call cursor.description
+            # to force submit statement immediately.
+            cursor.description  # noqa
```

Comment:
Because it'll fails in this scenario
```python
cur = connection.execute(text("UPDATE purchases SET status = 'OVERDUE' WHERE ship_date IS NULL;"))

# User may wanna get total number of records updated by call `cur.fetchall()`
# But it'll return null because cursor already fetchall in `do_execute` and ignore the result.
print(f"Total record updated {cur.fetchall()}")
```